src/db/service/ExecutionService.ts

```diff
@@ -1,11 +1,142 @@
 
 export default class ExecutionService {
   repository: Repository<ExecutionEntity>;
 
   constructor() {
     this.repository = AppDataSource.getRepository(ExecutionEntity);
   }
+
+  public async checkPassFail(
+    procedure: string,
+    parameters: string[],
+    test: ExecutionModelType
+  ) {
+    const tableRows: { [key: string]: number } = {};
+    test.rules.forEach((rule: RuleModelType) => {
+      rule.unitTests.forEach(async (unitTest: UnitTestType) => {
+        const { table } = unitTest;
+        tableRows.tableName = Number(
+          await store
+            .getState()
+            .connection.database.get(unitTest.rule.database)
+            ?.numRecordsInTable(table)
+        );
+      });
+    });
+    new Procedures().triggerProcedure(procedure, parameters);
+    test.rules.forEach(async (rule: RuleModelType) => {
+      rule.unitTests.forEach(async (unitTest: UnitTestType) => {
+        const { expectedRecordMatches, total, expectedNumRecords, table } =
+          unitTest;
+        switch (unitTest.level) {
+          case UnitTestOperations.TableGenericOperations: {
+            const res = await new TableTestService().check(
+              unitTest as TableTestType
+            );
+            if (unitTest.operation === TableGenericOperations.EXISTS) {
+              unitTest.result = Boolean(res);
+              unitTest.output = `${unitTest.table} table exists is ${unitTest.result}`;
+            } else {
+              switch (expectedRecordMatches) {
+                case RecordMatches.ZERO:
+                  unitTest.result = Number(res) === 0;
+                  unitTest.output = `${unitTest.table} had ${Number(
+                    res
+                  )} rows and expected 0 rows`;
+                  break;
+                case RecordMatches.GREATER_THAN:
+                  unitTest.result = Number(res) > 0;
+                  unitTest.output = `${unitTest.table} had ${Number(
+                    res
+                  )} rows and expected more than 0 rows`;
+                  break;
+                case RecordMatches.TABLE_ROWS:
+                  if (total) {
+                    unitTest.result = Number(res) === expectedNumRecords;
+                    unitTest.output = `${unitTest.table} had ${Number(
+                      res
+                    )} rows and expected ${expectedNumRecords} rows`;
+                  } else {
+                    unitTest.result =
+                      Number(res) - tableRows[table] === expectedNumRecords;
+                    unitTest.output = `${unitTest.table} had ${
+                      Number(res) - tableRows[table]
+                    } new rows and expected ${expectedNumRecords} newrows`;
+                  }
+                  break;
+                default:
+                  unitTest.result = false;
+                  break;
+              }
+            }
+            break;
+          }
+          case UnitTestOperations.RowBooleanOperations: {
```

Comment:
```suggestion
          case UnitTestOperations.RowStringOperations
          case UnitTestOperations.RowNumberOperations
          case UnitTestOperations.RowIDOperations
          case UnitTestOperations.RowBooleanOperations: {
```

Not sure if this is correct syntax wise but you should be able to do this. All of the row cases should trigger this code

Reply:
Oh yeah my bad I meant to do this for all of them